superdesk/io/commands/update_ingest.py

```diff
@@ -259,6 +259,27 @@ def process_anpa_category(item, provider):
         raise ProviderError.anpaError(ex, provider)
 
 
+def derive_category(item, provider):
+    """
+    Assuming that the item has at least one itpc subject use the vocabulary map to derive an anpa category
+    :param item:
+    :return: An item with a category if possible
+    """
+    try:
+        categories = []
+        map = superdesk.get_resource_service('vocabularies').find_one(req=None, _id='iptc_category_map')
+        if map:
+            for subject in item.get('subject', []):
+                for map_entry in map['items']:
```

Comment:
If you feel like it, you can simplify the double for loop with [itertools.product](https://docs.python.org/3/library/itertools.html#itertools.product), saving one level of indentation.